detectable/src/main/java/com/synopsys/integration/detectable/detectables/rubygems/gemlock/parse/GemlockParser.java

```diff
@@ -101,39 +102,36 @@ public DependencyGraph parseProjectDependencies(final List<String> gemfileLockLi
             }
         }
 
-        final List<String> missingDependencies = encounteredDependencies.stream().filter(it -> !resolvedDependencies.contains(it)).collect(Collectors.toList());
-        for (final String missingName : missingDependencies) {
+        List<String> missingDependencies = encounteredDependencies.stream().filter(it -> !resolvedDependencies.contains(it)).collect(Collectors.toList());
+        for (String missingName : missingDependencies) {
             final String missingVersion = "";
-            final DependencyId dependencyId = new NameDependencyId(missingName);
-            final ExternalId externalId = externalIdFactory.createNameVersionExternalId(Forge.RUBYGEMS, missingName, missingVersion);
+            DependencyId dependencyId = new NameDependencyId(missingName);
+            ExternalId externalId = externalIdFactory.createNameVersionExternalId(Forge.RUBYGEMS, missingName, missingVersion);
             lazyBuilder.setDependencyInfo(dependencyId, missingName, missingVersion, externalId);
         }
 
         return lazyBuilder.build();
     }
 
-    private void discoveredDependencyInfo(final NameVersionDependencyId id) {
-        final NameDependencyId nameOnlyId = new NameDependencyId(id.getName());
+    private void discoveredDependencyInfo(NameVersionDependencyId id) {
+        NameDependencyId nameOnlyId = new NameDependencyId(id.getName());
 
         //regardless we found the external id for this specific dependency.
-        final ExternalId externalId = externalIdFactory.createNameVersionExternalId(Forge.RUBYGEMS, id.getName(), id.getVersion());
+        ExternalId externalId = externalIdFactory.createNameVersionExternalId(Forge.RUBYGEMS, id.getName(), id.getVersion());
         lazyBuilder.setDependencyInfo(id, id.getName(), id.getVersion(), externalId);
 
         if (!resolvedDependencies.contains(id.getName())) { //if this is our first time encountering a dependency of this name, we become the 'version-less'
             resolvedDependencies.add(id.getName());
             lazyBuilder.setDependencyInfo(nameOnlyId, id.getName(), id.getVersion(), externalId);
-        } else {//otherwise, add us as a child to the version-less
-            lazyBuilder.addChildWithParent(id, nameOnlyId);
         }
-
```

Comment:
The removal of the else where the addChildWithParent is removing the circular dependency that caused the issue.  We were adding a dependency with a version as a child to the dependency that only contains the name not the version.  When the dependency graph was built it would have the versioned dependency have a link to itself.